bagitobjecttransfer/recordtransfer/forms/transfer_forms.py

```diff
@@ -902,4 +1006,137 @@ class ReviewForm(TransferForm):
     it.
     """
 
+    class Meta:
+        """Meta information for the form."""
+
+        transfer_step = TransferStep.REVIEW
+
     captcha = ReCaptchaField(widget=ReCaptchaV2Invisible, label="hidden")
+
+    @staticmethod
+    def _get_base_fields_data(form: BaseForm) -> dict[str, Any]:
+        """Extract fields data from a form."""
+        return {
+            form.fields[field].label or field: form.cleaned_data.get(field, "")
+            for field in form.fields
+            if form.fields[field].label != "hidden"
+        }
+
+    @staticmethod
+    def _process_file_upload(form: UploadFilesForm, user: User) -> dict[str, Any]:
+        """Handle file upload specific field processing."""
+        fields_data = ReviewForm._get_base_fields_data(form)
+        session_token = form.cleaned_data.get("session_token")
+
+        if not (session_token and user):
+            return fields_data
+
+        session = UploadSession.objects.filter(token=session_token, user=user).first()
+        if not session:
+            return fields_data
+
+        # Adds on links to access uploaded files
+        fields_data["Uploaded files"] = [
+            {"name": f.name, "url": f.get_file_access_url()}
+            for f in session.get_temporary_uploads()
+        ]
+
+        return fields_data
+
+    @staticmethod
+    def _process_group_transfer(form: GroupTransferForm, user: User) -> dict[str, Any]:
+        """Handle group transfer specific field processing."""
+        fields_data = ReviewForm._get_base_fields_data(form)
+        group_id = form.cleaned_data.get("group_id")
+
+        if not (group_id and user):
+            return fields_data
+
+        group = SubmissionGroup.objects.filter(created_by=user, uuid=group_id).first()
+        if not group:
+            return fields_data
+
+        # Replaces group UUID with group name
+        group_id_label = form.fields["group_id"].label
+        if group_id_label:
+            fields_data[group_id_label] = group.name
+
+        return fields_data
+
+    @staticmethod
+    def _process_formset(form: BaseFormSet) -> tuple[list[dict[str, Any]], Optional[str]]:
```

Comment:
To make consistent with `_get_base_fields_data`:

```suggestion
    def _get_formset_fields_data(form: BaseFormSet) -> tuple[list[dict[str, Any]], Optional[str]]:
```

Reply:
Addressed by:
https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/329/commits/6b727ae5950d5196799997dba3336eee36f904d7